### PR TITLE
Added test for ChangeLogModalController, AppFooterController and PeerSelectController

### PR DESCRIPTION
**Pull request by bartist**
Part of Issue #1354.
This pull request will improve the test scores of the controller.js as follows:

|             | % Statements | % Branch | % Functions | % Lines |
|:-----------:|:------------:|:--------:|:-------------:|:---------:|
|  Currently  |      3.5     |     0    |    0.36     |   3.52  |
| After Merge |      5.61    |    1.16  |    2.88     |   5.64  |

This pull request fully tests the ChangeLogModalController, AppFooterController and PeerSelectController.
If a certain style is demanded for creating tests, then we will adapt the test to conform with the demanded style.
This is also the reason that this pull request tests some of the smaller controllers, in case changes are needed.

### CODE REVIEW

**Comment by bartist** [comment]
@Ryuno-Ki 
Since you added the initial libraries and test to the webogram project, could you also review the added tests?
Thank you in advance

**Comment by Ryuno-Ki** [comment]
Lemme see.

File: test/unit/AppFooterController.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        $scope = _$rootScope_.$new();
+        $controller('AppFooterController', {
+            $scope: $scope,
+            LayoutSwitchService: service
+        });
+    });
+  });
+
+  // define tests
+  it('compiles', function (done) {
```
Comment (Ryuno-Ki) [inline test/unit/AppFooterController.js]: Superfluous. It just consumes CPU time.

File: test/unit/AppFooterController.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+describe('AppFooterController', function() {
```
Comment (Ryuno-Ki) [inline test/unit/AppFooterController.js]: Below you have spaces before the argument braces. Could you stick to one code style?

File: test/unit/AppFooterController.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  var $controller, $scope, service, serviceFlag;
+
+  beforeEach(module('myApp.controllers'));
+
+  beforeEach(function () {
+    serviceFlag = false;
+    service = {
+      switchLayout: function(parameter){
```
Comment (Ryuno-Ki) [inline test/unit/AppFooterController.js]: Below you have a space before the arguments braces.

File: test/unit/AppFooterController.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  // define tests
+  it('compiles', function (done) {
+    expect(true).toBe(true);
+    done();
+  });
+
+  it('calls the right function', function (done) {
+    expect(serviceFlag).toBe(false);
+    $scope.switchLayout(null);
```
Comment (Ryuno-Ki) [inline test/unit/AppFooterController.js]: `switchLayout` sounds like a toggle. You should check the value after another call. I haven't checked the actual implementation.

File: test/unit/ChangelogModalControlelerSpec.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
+describe('ChangeLogModalController', function() {
```
Comment (Ryuno-Ki) [inline test/unit/ChangelogModalControlelerSpec.js]: Below you have a space before the arguments braces.

File: test/unit/PeerSelectControllerSpec.js [hunk 6]
```diff
@@ -0,0 +1,258 @@
+describe('PeerSelectController', function () {
+  var $controller, $scope, $q, $mod, $APManager, $EService;
+  var createController, timeoutTime;
+  var $promiseData, $promise, $promiseFlag;
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Why do you split it on three lines here?

File: test/unit/PeerSelectControllerSpec.js [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+      }
+    };
+
+    timeoutTime = 1000;
+
+    $APManager = {
+      getPeerString: function (str){
+        return "P".concat(str);
+      },
+      getPeerID: function(str) {
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Usually you have a space before the arguments braces.

File: test/unit/PeerSelectControllerSpec.js [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+        return str.slice(-1);
+      },
+      getPeer: function (id) {
+        return id.concat("peer");
+      }
+    };
+
+    // The controller is created in the test in order to test different initial content of scope variables.
+    createController = function() {
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Usually you have a space before the arguments braces.

File: test/unit/PeerSelectControllerSpec.js [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    $promiseFlag = false;
+    $promise = {
+      then: function (f) {
+        $promiseFlag = true;
+        f();
+      }
+    };
+
+    $EService = {
+      confirm: function(data){
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Usually you have a space before the arguments braces.

File: test/unit/PeerSelectControllerSpec.js [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+      }
+    };
+
+    inject(function (_$controller_, _$rootScope_) {
+      $controller = _$controller_;
+      $scope = _$rootScope_.$new();
+    });
+  });
+
+  it('compiles', function (done) {
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Maybe use 'initialises properly' as test description?

File: test/unit/PeerSelectControllerSpec.js [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+
+  it('compiles with a shareLinkPromise that resolves', function (done) {
+    $scope.shareLinkPromise = {
+      then: function (resolve, reject) {
+        setTimeout(resolve("test"), timeoutTime);
+      }
+    }
+    createController();
+
+    setTimeout(function(){
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Usually you have a space before arguments braces.

File: test/unit/PeerSelectControllerSpec.js [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+      expect($scope.selectedCount).toBeDefined();
+    }, timeoutTime);
+
+    done();
+  });
+
+  it('compiles with a shareLinkPromise that resolves', function (done) {
+    $scope.shareLinkPromise = {
+      then: function (resolve, reject) {
+        setTimeout(resolve("test"), timeoutTime);
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: I feel like you should move that string into a variable. I was surprised to find an 'url' "test" below.

File: test/unit/PeerSelectControllerSpec.js [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+
+  it('compiles with a shareLinkPromise that doesn\'t resolve', function (done) {
+    $scope.shareLinkPromise = {
+      then: function (resolve, reject) {
+        setTimeout(reject(), timeoutTime);
+      }
+    }
+    createController();
+
+    setTimeout(function(){
```
Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Usually you have a space before arguments braces.

Comment (Ryuno-Ki) [inline test/unit/PeerSelectControllerSpec.js]: Usually you have a space before arguments braces.

File: test/unit/AppFooterController.js (hunk 4 above)
Comment (bartist) [inline test/unit/AppFooterController.js]: The switchLayout function calls the switchLayout function of the LayoutSwitchService. The only concern of the test is to check if this function is called.
If the switchLayout function of the LayoutSwitchService is a toggle function, then that should be tested in the tests for the LayoutSwitchService
(for reference: controller.js, line 2980)

**Comment by bartist** [comment]
Thank you for your time!
I will look more careful with my code style

**Comment by zhukov** [comment]
Wow. Thanks for the PR @bartist

**Pull request merged**